test/verify/run-tests

```diff
@@ -1,18 +1,90 @@
 #!/usr/bin/python3
 
 import glob
-import imp
 import os
 import string
 import subprocess
 import sys
 import unittest
+import dataclasses
+import errno
+import fcntl
+import time
+
+import importlib.machinery
+import importlib.util
 
 import parent
 import testlib
 import testvm
 
 sys.dont_write_bytecode = True
+os.environ['PYTHONUNBUFFERED'] = '1'
+
+
+@dataclasses.dataclass
+class Test:
+    test_id: int
+    command: list
+    process: subprocess.Popen = None
+    retries: int = 0
+    output: bytes = b""
+
+
+def print_test(test, print_ok):
+    for line in test.output.splitlines():
```

Comment:
Use `keepends=True` to avoid having to build an entirely new string from a copy and the `b"\n"`. As this is quite a time sensitive operation, some optimization here can't hurt.